app/Main/Cabal.hs

```diff
@@ -40,13 +49,24 @@ cabalOptions = do
     , pure . CabalDontDiscover . CabalTestsUnit <$> strOption (long "no-discover-test" <> help "Don't discover specified test import dependencies")
     ]
   optIncludeExternal <- switch (long "include-external" <> help "Include external import dependencies")
+  optNumJobs <- optional (option auto (long "jobs" <> short 'j' <> help "Number of worker threads to use"))
+  optPruneTo <- many $ strOption (long "prune-to" <> help "Only discover import dependencies of the specified module(s)")
+  optPruneToRegex <- many $ strOption (long "prune-to-regex" <> help "Only discover import dependencies of modules that match a regex")
   pure CabalOptions{..}
 
 runCabal :: CabalOptions -> IO ()
 runCabal CabalOptions{..} = do
+  numCapabilities <- getNumCapabilities
+  let numJobs =fromMaybe numCapabilities optNumJobs
+  logit $ unwords ["Discovering with num jobs = ", show numJobs]
+
+  let pruneToExplicit = flip elem <$> nonEmpty optPruneTo
+  let pruneToRegex = (\patterns (ModuleName m) -> any (m =~) patterns) <$> nonEmpty optPruneToRegex
+  let pruneTo = alignWith (mergeThese (liftA2 (||))) pruneToExplicit pruneToRegex
```

Comment:
If you provide multiple modules or multiple regexes to prune, just one has to match.